### PR TITLE
yt-dlp 2021.08.10 (new formula)

### PR DESCRIPTION
**Pull request by herrriehm**
yt-dlp is a fork of youtube-dl. As such, this formula is heavily based on youtube-dl's formula.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Unfortunately, I cannot quite get `brew install --build-from-source yt-dlp` to not fail. But I don’t know why. Can anybody help me?

### CODE REVIEW

**Review by ctsrc — CHANGES_REQUESTED** [review]

File: Formula/yt-dlp.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    virtualenv_install_with_resources
+  end
+
+  test do
+    # commit history of homebrew-core repo
+    system "#{bin}/youtube-dl", "--simulate", "https://www.youtube.com/watch?v=pOtd1cbOP7k"
+    # homebrew playlist
+    system "#{bin}/youtube-dl", "--simulate", "--yes-playlist", "https://www.youtube.com/watch?v=pOtd1cbOP7k&list=PLMsZ739TZDoLj9u_nob8jBKSC-mZb0Nhj"
+  end
+end
```
[the remark this answers is not in the document]
Comment (herrriehm) [reply Formula/yt-dlp.rb]: @ctsrc thank you very much for your help!

File: Formula/yt-dlp.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class YtDlp < Formula
+  include Language::Python::Virtualenv
+
+  desc "Fork of youtube-dl with additional features and fixes"
+  homepage "https://discord.gg/H5MNcFW63r"
+  url "https://github.com/yt-dlp/yt-dlp/releases/download/2021.08.10/yt-dlp.tar.gz"
+  sha256 "15bd1028840e9230f7b89ee928914b119858b9011c7083c6cb0c5a60d525f1e6"
```
[the remark this answers is not in the document]
Comment (herrriehm) [reply Formula/yt-dlp.rb]: thanks for your help @cho-m! I added the changes of @ctsrc and got the formula working.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@herrriehm, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by nao20010128nao** [comment]
Just curiosity, how long does it take to reflect yt-dlp releases into core's yt-dlp formulae?

**Comment by ctsrc** [comment]
@nao20010128nao my experience is that pull requests for updated releases are merged quickly. I see that you guys over on yt-dlp started maintaining a homebrew tap. I think that it would be better to submit pull requests for updating the formula here in homebrew core instead. This way it’s one less step for users.

**Comment by ctsrc** [comment]
@nao20010128nao now that a couple of updated yt-dlp releases have been submitted and merged, you can see they are quite quick :)

See #84432 and #84438

**Pull request closed without merging**